switched.c

```diff
@@ -127,16 +172,26 @@ int main(int argc, char **argv) {
             return 2;
         }
     }
-    
-    cpu_t cpu = {.pc = 0, .sp = -1, .state = Cpu_Running, 
+
+    cpu_t cpu = {.pc = 0, .sp = -1, .csp = -1, .state = Cpu_Running,
                  .steps = 0, .stack = {0},
                  .pmem = Program};
-    
-    while (cpu.state == Cpu_Running && cpu.steps < steplimit) {
+
+    // First, handle pre-enty-point code (like exception handlers)
```

Comment:
This jump is kind of 'interrupt table', but for exceptions. The user will specify the address where to look for exception handlers. This address can not be a constant, as we store exception handler code in program memory, and we do not know neither the size of the program, nor the size of exception handler.